File: 693/BinaryNumberWithAlternatingBits.scala
object BinaryNumberWithAlternatingBits {
  def hasAlternatingBits(n: Int): Boolean = {
    def hasOnlyOneBit(n: Int): Boolean =
      (n & (n - 1)) == 0

    val t = n ^ (n >> 1)
    hasOnlyOneBit(t ^ (t >> 1))
  }

  def main(args: Array[String]): Unit = {
    val n = 5
    println(hasAlternatingBits(n))
  }
}

File: 693/binary-number-with-alternating-bits.js
/**
 * @param {number} n
 * @return {boolean}
 */
var hasAlternatingBits = function (n) {
    const hasOnlyOneBit = n => (n & (n - 1)) == 0;

    let t = n ^ (n >> 1);
    return hasOnlyOneBit(t ^ (t >> 1));
};